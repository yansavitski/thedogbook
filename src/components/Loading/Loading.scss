@keyframes rotateAnimation {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.Loading {
  text-align: center;
  width: 100%;

  &-spinner {
    animation: rotateAnimation 1.3s ease-out infinite;
  }
}
