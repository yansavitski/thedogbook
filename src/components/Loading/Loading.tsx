import './Loading.scss'

import classNames from 'classnames'
import React from 'react'
import { AiOutlineLoading3Quarters as Spinner } from 'react-icons/ai'

const DEFAULT_SIZE = 24

interface LoadingProps {
  className?: string
  size?: number
}

export const Loading: React.FunctionComponent<LoadingProps> = ({
  className,
  size = DEFAULT_SIZE,
}: LoadingProps) => (
  <div className={classNames('Loading', className)}>
    <Spinner className="Loading-spinner" size={size} />
  </div>
)
