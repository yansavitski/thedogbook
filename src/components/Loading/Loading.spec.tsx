import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'

import { Loading } from './Loading'

describe('<Loading />', () => {
  it('renders correctly', () => {
    expect.assertions(1)
    const tree = shallow(<Loading className="customLoader" size={32} />)

    expect(toJson(tree)).toMatchSnapshot()
  })
})
