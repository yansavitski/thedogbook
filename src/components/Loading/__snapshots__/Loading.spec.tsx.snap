// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders correctly 1`] = `
<div
  className="Loading customLoader"
>
  <AiOutlineLoading3Quarters
    className="Loading-spinner"
    size={32}
  />
</div>
`;
